libs/shared/flagd-core/src/lib/storage.ts

```diff
@@ -12,6 +12,14 @@ export interface Storage {
    */
   setConfigurations(cfg: string): void;
 
+  /**
+   * Updates the configurations and returns the list of flags that have changed.
+   * @param cfg The configuration string to be parsed and stored.
+   * @returns The list of flags that have changed.
+   * @throws {Error} If the configuration string is invalid.
+   */
+  updateConfigurations(cfg: string): string[];
```

Comment:
Would it be a problem to simply add this functionality to `setConfigurations`?

Reply:
I went back and forth on this. I'm happy to update `setConfigurations`.

Reply:
From a UX standpoint I think it reduces confusion to do that.